### PR TITLE
Implemented to use Intl.PluralRules object in IString if it's available

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [ ] `ReleaseNotes` has added or is not needed.
* [ ] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Implemented to use Intl.PluralRules object in IString if it's available.
* test expectation values is written in cldr 41
  *  https://unicode-org.github.io/cldr-staging/charts/41/supplemental/language_plural_rules.html
* tested various versions of the node through circleCI build.
* tested chrome and firefox browsers.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/test/root/teststrings.js [hunk 1]
```diff
@@ -3244,9 +3270,10 @@ module.exports.teststrings = {
         test.ok(str !== null);
 
         if (ilib._getPlatform() === "nodejs") {
-            var version = process.versions["node"];
-            var majorVersion = version.split(".")[0];
-            if (Number(majorVersion) < 16) {
+            var cldrVersion = process.versions["cldr"];
+            if (Number(cldrVersion) < 36) { // Intl.PluralRules doesn't support this locale until this version.
+                test.equal(str.formatChoice(1000000), "The items are many");
+            } else if (Number(cldrVersion) >= 36 && Number(cldrVersion) < 40 ) {
```
Comment (ehoogerbeets) [inline js/test/root/teststrings.js:3276]: Should that be `Number(cldrVersion) <= 40`

File: js/test/root/teststrings.js [hunk 2]
```diff
@@ -3863,9 +3917,10 @@ module.exports.teststrings = {
         test.ok(str !== null);
 
         if (ilib._getPlatform() === "nodejs") {
-            var version = process.versions["node"];
-            var majorVersion = version.split(".")[0];
-            if (Number(majorVersion) < 16) {
+            var cldrVersion = process.versions["cldr"];
+            if (Number(cldrVersion) < 36) { // // Intl.PluralRules doesn't support this locale until this version.
+                test.equal(str.formatChoice(3e6), "The items are many");
+            } else if (Number(cldrVersion) >= 36 && Number(cldrVersion) < 40 ) {
```
Comment (ehoogerbeets) [inline js/test/root/teststrings.js:3923]: And here? Or was the change in 40, not 41?

File: js/test/root/teststrings.js (hunk 1 above)
Comment (gouniLee) [reply js/test/root/teststrings.js:3276]: It changed to version 40. https://unicode-org.github.io/cldr-staging/charts/40/supplemental/language_plural_rules.html#pt

**Comment by gouniLee** [comment]
@ehoogerbeets It is ready to get your review.

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**